src/core/lwip/tcp_out.c

```diff
@@ -748,6 +749,12 @@ err_t tcp_write_express(struct tcp_pcb *pcb, const struct iovec *iov, u32_t iovc
     }
     last_seglen = last ? last->len : 0;
 
+    if (desc->attr == PBUF_DESC_EXPRESS) {
+        /* Keep opaque value only in the right most pbuf for each send operation. */
+        opaque = desc->opaque;
+        desc->opaque = NULL;
```

Comment:
Consider adding here a comment why we null the desc->opaque and then restore it.